src/webui/www/private/scripts/file-tree.js

```diff
@@ -28,149 +28,163 @@
 
 'use strict';
 
-const FilePriority = {
-    "Ignored": 0,
-    "Normal": 1,
-    "High": 6,
-    "Maximum": 7,
-    "Mixed": -1
-};
-Object.freeze(FilePriority);
-
-const TriState = {
-    "Unchecked": 0,
-    "Checked": 1,
-    "Partial": 2
-};
-Object.freeze(TriState);
-
-const FileTree = new Class({
-    root: null,
-    nodeMap: {},
-
-    setRoot: function(root) {
-        this.root = root;
-        this.generateNodeMap(root);
-
-        if (this.root.isFolder)
-            this.root.calculateSize();
-    },
-
-    getRoot: function() {
-        return this.root;
-    },
-
-    generateNodeMap: function(node) {
-        // don't store root node in map
-        if (node.root !== null) {
-            this.nodeMap[node.rowId] = node;
-        }
-
-        node.children.each(function(child) {
-            this.generateNodeMap(child);
-        }.bind(this));
-    },
-
-    getNode: function(rowId) {
-        return (this.nodeMap[rowId] === undefined)
-            ? null
-            : this.nodeMap[rowId];
-    },
-
-    getRowId: function(node) {
-        return node.rowId;
-    },
-
-    /**
-     * Returns the nodes in dfs order
-     */
-    toArray: function() {
-        const nodes = [];
-        this.root.children.each(function(node) {
-            this._getArrayOfNodes(node, nodes);
-        }.bind(this));
-        return nodes;
-    },
-
-    _getArrayOfNodes: function(node, array) {
-        array.push(node);
-        node.children.each(function(child) {
-            this._getArrayOfNodes(child, array);
-        }.bind(this));
-    }
-});
-
-const FileNode = new Class({
-    name: "",
-    rowId: null,
-    size: 0,
-    checked: TriState.Unchecked,
-    remaining: 0,
-    progress: 0,
-    priority: FilePriority.Normal,
-    availability: 0,
-    depth: 0,
-    root: null,
-    data: null,
-    isFolder: false,
-    children: [],
-});
-
-const FolderNode = new Class({
-    Extends: FileNode,
-
-    initialize: function() {
-        this.isFolder = true;
-    },
-
-    addChild(node) {
-        this.children.push(node);
-    },
-
-    /**
-     * Recursively calculate size of node and its children
-     */
-    calculateSize: function() {
-        let size = 0;
-        let remaining = 0;
-        let progress = 0;
-        let availability = 0;
-        let checked = TriState.Unchecked;
-        let priority = FilePriority.Normal;
-
-        let isFirstFile = true;
-
-        this.children.each(function(node) {
-            if (node.isFolder)
-                node.calculateSize();
-
-            size += node.size;
-
-            if (isFirstFile) {
-                priority = node.priority;
-                checked = node.checked;
-                isFirstFile = false;
-            }
-            else {
-                if (priority !== node.priority)
-                    priority = FilePriority.Mixed;
-                if (checked !== node.checked)
-                    checked = TriState.Partial;
+window.FileTree = (function() {
```

Comment:
Is it common or necessary to use the `window` object? 
[Here](https://stackoverflow.com/questions/12743007/can-i-add-attributes-to-window-object-in-javascript) says it is "frowned upon" and I'm inclined to think so.

Reply:
Moved all modules under window.qBittorrent.